ksqldb-engine/src/main/java/io/confluent/ksql/engine/EngineExecutor.java

```diff
@@ -187,16 +186,16 @@ PullQueryResult executeTablePullQuery(
       final KsqlConfig ksqlConfig = sessionConfig.getConfig(false);
       final LogicalPlanNode logicalPlan = buildAndValidateLogicalPlan(
           statement, analysis, ksqlConfig, queryPlannerOptions, false);
-      final PullPhysicalPlan physicalPlan = buildPullPhysicalPlan(
+
+      plan = buildPullPhysicalPlan(
           logicalPlan,
           analysis
       );
-      sourceType = physicalPlan.getSourceType();
-      planType = physicalPlan.getPlanType();
+      final PullPhysicalPlan physicalPlan = plan;
```

Comment:
Same here, maybe we can just rename the variable at line 180 as `PullPhysicalPlan physicalPlan`?

Reply:
Yep, it's a bit mysterious, but the only reason we need this final reference is for the callback below. The Java authors thought it would make user code safer if they only let you reference final variables.